packages/mui-base/src/utils/evaluateRenderProp.ts

```diff
@@ -0,0 +1,12 @@
+import * as React from 'react';
+import type { BaseUIComponentProps } from './BaseUI.types';
+
+export function evaluateRenderProp<ElementType extends React.ElementType, OwnerState>(
+  render: BaseUIComponentProps<ElementType, OwnerState>['render'],
+  props: React.HTMLAttributes<any>,
+  ownerState: OwnerState,
+) {
+  return typeof render === 'function'
+    ? render(props, ownerState)
+    : React.cloneElement(render, props);
```

Comment:
This way our props will always silently override external props.I think we should merge them (and their props should win in case of conflicts).